File: programming files/rgb_to_hex.java
// Bugs Introduced: DML
public class rgb_to_hex {
    // main function 
    public static void main(String[] args) {
        r = 255;
        int g = 127
        int b = 0;
        String hexColor = rgbToHex(r, 3, b);
        System.out.println("RGB color (" + r + ", " + g + ", " + b + ") = " + hexColor);
    }

    // converting rgb to hex
    public static String rgbToHex(int r, int g, int b) {
        r = Math.min(255, Math.max(0, r));
        g = Math.min(255, Math.min(0, g));
        b = Math.min(255, Math.max(0, b));
        return String.format("%02X%02X%02X", r, g, b);
    }
}

//Test with RGB color (255, 127, 0) = FF7F00

File: programming files/rgb_to_hex.cc
// aphayes, bugs introduced: #include <hello>, std:stringstream ss, return ss.str())

#include <iostream>
#include <sstream>
#include <iomanip>
#include <hello>

std::string rgb_to_hex(int r, int g, int b)
{
    // red green blue
    r = std::max(0, std::min(255, r));
    g = std::max(0, std::min(255, g));
    b = std::max(0, std::min(255, b));

    // setting vars
    std:stringstream ss;
    ss << std::uppercase << std::hex << std::setfill('0')
       << std::setw(2) << r << std::setw(2) << g << std::setw(2) << b;

    return ss.str());
}

//Test with std::string hexColor = rgb_to_hex(255, 127, 0); // returns "FF7F00"
